src/XRP/common-xrpl-client.ts

```diff
@@ -0,0 +1,277 @@
+/* eslint-disable no-restricted-syntax */
+import { Signer, Utils, Wallet, XrplNetwork } from 'xpring-common-js'
+import XrpUtils from './xrp-utils'
+import GrpcNetworkClient from './grpc-xrp-network-client'
+import GrpcNetworkClientWeb from './grpc-xrp-network-client.web'
+import { XRPDropsAmount } from './Generated/web/org/xrpl/rpc/v1/amount_pb'
+import { AccountRoot } from './Generated/web/org/xrpl/rpc/v1/ledger_objects_pb'
+import {
+  Account,
+  LastLedgerSequence,
+  SigningPublicKey,
+} from './Generated/web/org/xrpl/rpc/v1/common_pb'
+import { Transaction } from './Generated/web/org/xrpl/rpc/v1/transaction_pb'
+import { AccountAddress } from './Generated/web/org/xrpl/rpc/v1/account_pb'
+import { GetFeeResponse } from './Generated/web/org/xrpl/rpc/v1/get_fee_pb'
+import TransactionStatus from './transaction-status'
+import RawTransactionStatus from './raw-transaction-status'
+import { XrpNetworkClient } from './xrp-network-client'
+import XrpError from './xrp-error'
+import { LedgerSpecifier } from './Generated/web/org/xrpl/rpc/v1/ledger_pb'
+import TransactionResult from './model/transaction-result'
+import isNode from '../Common/utils'
+
+/** A margin to pad the current ledger sequence with when submitting transactions. */
+const maxLedgerVersionOffset = 10
+
+/**
+ * CommonXrplClient is a client which interacts with the XRP Ledger.
+ */
+export default class CommonXrplClient {
+  /**
+   * Create a new CommonXrplClient.
+   *
+   * The CommonXrplClient will use gRPC to communicate with the given endpoint.
+   *
+   * @param grpcUrl The URL of the gRPC instance to connect to.
+   * @param network The network this XrpClient is connecting to.
+   * @param forceWeb If `true`, then we will use the gRPC-Web client even when on Node. Defaults to false. This is mainly for testing and in the future will be removed when we have browser testing.
+   */
+  public static commonXrplClientWithEndpoint(
+    grpcUrl: string,
+    network: XrplNetwork,
+    forceWeb = false,
+  ): CommonXrplClient {
+    return isNode() && !forceWeb
+      ? new CommonXrplClient(new GrpcNetworkClient(grpcUrl), network)
+      : new CommonXrplClient(new GrpcNetworkClientWeb(grpcUrl), network)
+  }
+
+  /**
+   * Create a new CommonXrplClient with a custom network client implementation.
+   *
+   * @param networkClient A network client which will manage remote RPCs to Rippled.
+   * @param network The network this XrpClient is connecting to.
+   */
+  public constructor(
+    public readonly networkClient: XrpNetworkClient,
+    readonly network: XrplNetwork,
+  ) {}
+
+  public async getOpenLedgerSequence(): Promise<number> {
```

Comment:
can we doc this and the other public functions?

Reply:
mm, yep, meant to do that. doc'ed